tests/CommunityToolkit.Mvvm.UnitTests/Test_ObservablePropertyAttribute.cs

```diff
@@ -360,6 +361,105 @@ public void Test_OnPropertyChangingAndChangedPartialMethodWithAdditionalValidati
         Assert.AreEqual("B", model.Name);
     }
 
+#if NET6_0_OR_GREATER
+    // See https://github.com/CommunityToolkit/dotnet/issues/155
+    [TestMethod]
+    public void Test_ObservableProperty_NullabilityAnnotations_Simple()
+    {
+        // List<string?>?
+        NullabilityInfoContext context = new();
+        NullabilityInfo info = context.Create(typeof(NullableRepro).GetProperty(nameof(NullableRepro.NullableList))!);
+
+        Assert.AreEqual(typeof(List<string>), info.Type);
+        Assert.AreEqual(NullabilityState.Nullable, info.ReadState);
+        Assert.AreEqual(NullabilityState.Nullable, info.WriteState);
+        Assert.AreEqual(1, info.GenericTypeArguments.Length);
+
+        NullabilityInfo elementInfo = info.GenericTypeArguments[0];
+
+        Assert.AreEqual(typeof(string), elementInfo.Type);
+        Assert.AreEqual(NullabilityState.Nullable, elementInfo.ReadState);
+        Assert.AreEqual(NullabilityState.Nullable, elementInfo.WriteState);
+    }
+
+    // See https://github.com/CommunityToolkit/dotnet/issues/155
+    [TestMethod]
+    public void Test_ObservableProperty_NullabilityAnnotations_Complex()
+    {
+        // Foo<Foo<string?, int>.Bar<object?>?, StrongBox<Foo<int, string?>.Bar<object>?>?>?
+        NullabilityInfoContext context = new();
+        NullabilityInfo info = context.Create(typeof(NullableRepro).GetProperty(nameof(NullableRepro.NullableMess))!);
+
+        Assert.AreEqual(typeof(Foo<Foo<string?, int>.Bar<object?>?, StrongBox<Foo<int, string?>.Bar<object>?>?>), info.Type);
+        Assert.AreEqual(NullabilityState.Nullable, info.ReadState);
+        Assert.AreEqual(NullabilityState.Nullable, info.WriteState);
+        Assert.AreEqual(2, info.GenericTypeArguments.Length);
+
+        NullabilityInfo leftInfo = info.GenericTypeArguments[0];
+
+        Assert.AreEqual(typeof(Foo<string?, int>.Bar<object?>), leftInfo.Type);
```

Comment:
Since this is such a large method, it might be helpful to add a few comments to visualize which part of the generic is being tested. It's not immediately obvious with the current variable names.